backend/danswer/connectors/gitlab/connector.py

```diff
@@ -0,0 +1,174 @@
+
```

Comment:
Could you kindly use the linter/formatter we use standard in the repo? But if this is too much trouble, not a big deal.

The mypy however is a bit more annoying for me to fix myself so could you please make sure the mypy checks pass?